pkg/pillar/types/misc.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"reflect"
+	"strings"
+)
+
+// ParseFunc is a function which takes a json string and produes
+// a specific type.
+type ParseFunc func(sb []byte) (interface{}, error)
```

Comment:
As far as I can tell, this only ever is used inside package `pubsub/`. Is that correct? If so, it probably should just be moved inside there and non-exported. Or is it intended to be used elsewhere?

Reply:
I agree.. Thisis causing PubSub to depend on types. This should be moved inside types.